src/main/java/de/tum/cit/aet/artemis/iris/service/settings/IrisSettingsService.java

```diff
@@ -276,9 +300,135 @@ private IrisCourseSettings updateCourseSettings(IrisCourseSettings existingSetti
         ));
         // @formatter:on
 
+        // Automatically update the exercise settings when the enabledForCategories is changed
+        var newEnabledForCategoriesExerciseChat = existingSettings.getIrisChatSettings() == null ? new TreeSet<String>()
+                : existingSettings.getIrisChatSettings().getEnabledForCategories();
+        if (!Objects.equals(oldEnabledForCategoriesExerciseChat, newEnabledForCategoriesExerciseChat)) {
```

Comment:
```suggestion
        if (!oldEnabledForCategoriesExerciseChat.equals(newEnabledForCategoriesExerciseChat)) {
```
You already have null safety